src/Type/KeyOfType.php

```diff
@@ -41,7 +46,7 @@ public function describe(VerbosityLevel $level): string
 
 	public function isResolvable(): bool
 	{
-		return !TypeUtils::containsTemplateType($this->type);
+		return !TypeUtils::containsTemplateType($this->type) && $this->type->isIterable()->yes();
```

Comment:
This isn't necessary, right? For example ArrayAccess has "keys" but isn't iterable unless paired with Traversable.